src/main/scala/is/hail/variant/VariantSampleMatrix.scala

```diff
@@ -145,8 +142,36 @@ object VSMSubgen {
     tGen = Genotype.genDosage, isDosage = true)
 }
 
-class VariantSampleMatrix[T](val hc: HailContext, val metadata: VariantMetadata,
-  val rdd: OrderedRDD[Locus, Variant, (Annotation, Iterable[T])])(implicit tct: ClassTag[T]) extends JoinAnnotator {
+class VariantSampleMatrix[T](val hc: HailContext, val metadata: VSMMetadata,
+  val ast: MatrixAST[T])(implicit val tct: ClassTag[T]) extends JoinAnnotator {
+
+  def this(hc: HailContext,
+    metadata: VSMMetadata,
+    localValue: VSMLocalValue,
+    rdd: OrderedRDD[Locus, Variant, (Annotation, Iterable[T])])(implicit tct: ClassTag[T]) =
+    this(hc, metadata,
+      MatrixLiteral(
+        MatrixType(metadata),
+        MatrixValue(localValue, rdd)))
+
+  def this(hc: HailContext, fileMetadata: VSMFileMetadata,
+    rdd: OrderedRDD[Locus, Variant, (Annotation, Iterable[T])])(implicit tct: ClassTag[T]) =
+    this(hc, fileMetadata.metadata, fileMetadata.localValue, rdd)
+
+  lazy val value = {
+    val opt = MatrixAST.optimize(ast)
+    println("ast", ast)
+    println("opt", opt)
```

Comment:
I noticed this was a little noisy too, should we make these log statements or perhaps remove them?